ste/jobStatusManager.go

```diff
@@ -61,15 +65,36 @@ func (jm *jobMgr) statusMgrClosed() bool {
 
 /* These functions should not fail */
 func (jm *jobMgr) SendJobPartCreatedMsg(msg JobPartCreatedMsg) {
-	jm.jstm.partCreated <- msg
-	if msg.IsFinalPart {
-		// Inform statusManager that this is all parts we've
-		close(jm.jstm.partCreated)
+	if jm.jstm.partCreated != nil { // Sends not allowed if channel is closed
+		jm.jstm.flagMutex.RLock()
```

Comment:
Channel push and pop are locked internally. Are we just trying to safeguard against parallel closure of the channel here ?

Reply:
We're trying to safeguard against panics from sending a message on a closed channel.  That might have been closed by another concurrent operation or go routine. Wrapping channel sends and closures with mutexes allows exclusive access to those parts of the code. What do you mean by locked internally?

Reply:
Not a problem! I'll replace the Mutex locking and unlocking mechanisms with this combo of `defer` and `recover` for channel sends